automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/PublishedWorkspaceSpec.scala

```diff
@@ -0,0 +1,130 @@
+
+              val clonedWorkspaceName = uuidWithPrefix("cloneWorkspace")
+
+              register cleanUp Orchestration.workspaces.delete(billingProject, clonedWorkspaceName)
+              Orchestration.workspaces.clone(billingProject, workspaceName, billingProject, clonedWorkspaceName)
+              // check cloned workspace does not have published status in workspace detail: "library:published": false
+              val response = Rawls.workspaces.getWorkspaceDetails(billingProject, clonedWorkspaceName)
+              response should include (""""library:published":false""")
+
+              // check cloned workspace is not visible in library
+              withClue("a cloned workspace should not be visible in library.") {
+                eventually {
+                  isVisibleInLibrary(billingProject, clonedWorkspaceName) shouldBe false
+                }
+              }
+
+            }
+          }
+        }
+
+        "should default to visible to 'all users' in Library Attributes" in {
+          val curatorUser = UserPool.chooseCurator
+          implicit val curatorAuthToken: AuthToken = curatorUser.makeAuthToken()
+
+          withCleanBillingProject(curatorUser) { billingProject =>
+            withWorkspace(billingProject, "PublishedWorkspaceSpec_workspace") { workspaceName =>
+
+              val data = LibraryData.metadataBasic + ("library:datasetName" -> workspaceName)
+              Orchestration.library.setLibraryAttributes(billingProject, workspaceName, data)
+              Orchestration.library.setDiscoverableGroups(billingProject, workspaceName, List("all_broad_users"))
+              Orchestration.library.publishWorkspace(billingProject, workspaceName)
+
+              val clonedWorkspaceName = workspaceName + "_clone"
+              register cleanUp Orchestration.workspaces.delete(billingProject, clonedWorkspaceName)
+              Orchestration.workspaces.clone(billingProject, workspaceName, billingProject, clonedWorkspaceName)
+
+              //Verify default group "All users"
+              //In swagger you make sure that getDiscoverableGroup endpoint shows []
+              withClue(s"Get api/library/${billingProject}/${clonedWorkspaceName}/discoverableGroups endpoint shows []") {
+                eventually {
+                  val accessGroup: Seq[String] = Orchestration.library.getDiscoverableGroups(billingProject, clonedWorkspaceName)
+                  accessGroup.size shouldBe 0
+                }
+              }
+            }
+          }
+        }
+
+      }
+
+    }
+  }
+
+
+  /**
+    *
+    * @return True: workspace is visible in library table
+    */
+  private def isVisibleInLibrary(namespace: String, workspaceName: String)(implicit token: AuthToken): Boolean = {
+    val response = Orchestration.library.searchPublishedLibraryDataset(namespace, workspaceName)
+
+    import DefaultJsonProtocol._
+    val total = JsonParser(response).asJsObject.fields("total").convertTo[Int]
+    total == 1
```

Comment:
Could Orch ever return more than one result for this search and should the test fail if it does?

Reply:
no. bc param search string is the `workspaceName`.